core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -19,12 +19,13 @@ trait RTS {
    * error, running forever, or producing an `A`.
    */
   final def unsafeRun[E, A](io: IO[E, A]): A = unsafeRunSync(io) match {
-    case ExitResult.Completed(v)  => v
-    case ExitResult.Terminated(t) => throw t
-    case ExitResult.Failed(e)     => throw Errors.UnhandledError(e)
+    case ExitResult.Completed(v)       => v
+    case ExitResult.Terminated(Nil)    => throw new Error("Fiber interrupted")
+    case ExitResult.Terminated(t :: _) => throw t
```

Comment:
Now `unsafeRun` should return the "first" error. So whether this is correct or not depends on the order in which the `Throwable` are stored. Most likely they would be stored in reverse order (first exception is at the end of the list), in which case this code is not correct.